pkg/plugins/manager/installer.go

```diff
@@ -130,6 +133,28 @@ func (m *PluginInstaller) Add(ctx context.Context, pluginID, version string, opt
 		return err
 	}
 
+	pluginErrs := m.pluginErrorResolver.PluginErrors()
+	if len(pluginErrs) > 0 {
+		for _, pluginErr := range pluginErrs {
+			if pluginErr.PluginID == pluginID {
+				// There is an error related to the plugin so uninstall it
+				err := m.pluginLoader.Unload(ctx, pluginID)
+				if err != nil {
+					if errors.Is(err, plugins.ErrPluginNotInstalled) {
+						// The plugin was not fully installed, remove the plugin files
+						remerr := os.RemoveAll(extractedArchive.Path)
```

Comment:
I am curious if we can do this @wbrowne. It's not strictly necessary but if we don't, the plugin files remain in the server and it will try to load it every time it starts. WDYT?